test/NuGet.Core.FuncTests/NuGet.Commands.FuncTest/RestoreCommandTests.cs

```diff
@@ -2023,11 +2027,14 @@ public async Task RestoreCommand_RestoreFloatingVersionWithFailingLocalSourceAsy
                     LockFilePath = Path.Combine(projectDir, "project.lock.json")
                 };
 
+                // Act & Assert
                 var command = new RestoreCommand(request);
+                var result = await command.ExecuteAsync();
+                await result.CommitAsync(logger, CancellationToken.None);
 
-                // Act & Assert
-                var ex = await Assert.ThrowsAsync<FatalProtocolException>(async () => await command.ExecuteAsync());
-                Assert.NotNull(ex);
+                Assert.False(result.Success);
+                Assert.Equal(1, logger.ErrorMessages.Count());
```

Comment:
If you agree with my other comment, then this is also fragile.

Reply:
i now recall. believe this is TestLogger...i don't believe NUxxxx errors get collected today. let me go look to see how hard...

Reply:
yes, TestLogger wasn't calling message.FormatWithCode() like Console class did.
I'm still doing string checks, but checking for NUxxxx is more robust.
thanks for suggesting.

Reply:
fixed now.